source/faq.txt

```diff
@@ -148,6 +146,33 @@ When ``MongoClient.close()`` is called by any thread, the driver
 closes all idle sockets and closes all sockets that are in
 use as they are returned to the pool.
 
+
+How do I prevent the "IllegalArgumentException: Invalid BSON field name" error?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your application may throw this exception when it encounters any of the
+following issues:
+
+- Characters in field names that your MongoDB deployment cannot process
```

Comment:
Have you verified this?

Reply:
Sorry, I did not because I assumed I had the information I needed from the related ticket and server manual. After testing, found out that I couldn't generate the error for this case it on MongoDB v4.2.